### PR TITLE
API: Vector::from_fn

### PR DESCRIPTION
**Pull request by sinhrks**
``Vector`` version of ``Matrix::from_fn``.

### CODE REVIEW

**Comment by AtheMathmo** [comment]
Good idea! It's good to see the `vector` module getting some love.

I think this PR looks ready to merge. I'll hang on a little to see if anyone else has any comments.

Thanks.

File: src/vector.rs [hunk 1]
```diff
@@ -43,6 +43,31 @@ impl<T> Vector<T> {
         }
     }
 
+    /// Constructor for Vector struct that takes a function `f`
+    /// and constructs a new vector such that `V_i = f(i)`,
+    /// where `i` is the index.
+    ///
+    /// Requires the vector data.
+    ///
```
Comment (theotherphil) [inline src/vector.rs]: Nitpick: I don't know what "Requires the vector data" means - I think this sentence should just be removed.

File: src/vector.rs [hunk 2]
```diff
@@ -825,6 +850,25 @@ mod tests {
     }
 
     #[test]
+    fn create_vector_new_from_fn() {
+        let v1 = Vector::from_fn(3, |x| x + 1);
```
Comment (theotherphil) [inline src/vector.rs]: Even nitpickier - should this be called create_vector_from_fn?

File: src/vector.rs (hunk 1 above)
Comment (AtheMathmo) [inline src/vector.rs]: I agree with this comment. In fact the advantage of this function is that it **does_not** require the vector data.

File: src/vector.rs (hunk 2 above)
Comment (AtheMathmo) [inline src/vector.rs]: I think `from_fn` is better for a few reasons.

- We already have a `from_fn` for matrices
- It is shorter
- It starts with `from` which is conventional
- There is some precedence for this function. There used to be a `from_fn` for `std::Vec`, and it does exist [elsewhere in the ecosystem](http://www.piston.rs/image/image/struct.ImageBuffer.html#method.from_fn)

**Comment by AtheMathmo** [comment]
Thank you for reviewing @theotherphil !

File: src/vector.rs (hunk 2 above)
Comment (theotherphil) [inline src/vector.rs]: I meant the name of the test!

Comment (AtheMathmo) [inline src/vector.rs]: oh, ha! In which case I agree :D

**Comment by sinhrks** [comment]
Thx, fixed the points.

**Comment by AtheMathmo** [comment]
Looks merge ready to me. Will give 24 hours for other comments then merge.

**Pull request merged**